### ISSUE TITLE
No license

### ISSUE THREAD

**Issue by jpetso** [nimbus-network/minimalt#1]
While this might be intended as a research project, adoption and potential continuation of development of this project could benefit from having a license added. Many projects on GitHub contain this as LICENSE.txt and add specific copyright holder information in the per-file copyright headers. Without a license, the code cannot legally be adopted into other sources by anyone except the copyright holder.

For a shared/static library like this, I imagine that appropriate licenses would be either of MIT, BSD, Apache 2.0 or LGPL, depending on how much copyleft is desired.

**Comment by gitbugged** [comment]
Agreed, would be nice to know what license this is in so that it can be forked or expanded upon. Ping @balihoo-anewman @rovaughn

**Comment by T12z** [comment]
> Agreed, would be nice to know what license this is in so that it can be forked or expanded upon.

Better don't.

Even though it was a nice approach, Minimalt does have security shortcomings in its concept - as far as I remember no PFS / ephemeral keys, etc. It is more of a proof of concept, not a real-world implementation.
Have a look at Wireguard and fork that to what you want. I know, the kernel implementation looks daunting, but there is also a simple handshake example in Go and complete userspace implementations somewhere.
